File: backend/SApplauz.Shared/DTOs/CreatePaymentIntentResponse.cs
namespace SApplauz.Shared.DTOs;

public class CreatePaymentIntentResponse
{
    public string ClientSecret { get; set; } = string.Empty;
    public string PaymentIntentId { get; set; } = string.Empty;
    public string PublishableKey { get; set; } = string.Empty;
}

File: packages/sapplauz_core/lib/models/create_payment_intent_response.dart
class CreatePaymentIntentResponse {
  final String clientSecret;
  final String paymentIntentId;
  final String publishableKey;

  CreatePaymentIntentResponse({
    required this.clientSecret,
    required this.paymentIntentId,
    required this.publishableKey,
  });

  factory CreatePaymentIntentResponse.fromJson(Map<String, dynamic> json) {
    return CreatePaymentIntentResponse(
      clientSecret: json['clientSecret']?.toString() ?? '',
      paymentIntentId: json['paymentIntentId']?.toString() ?? '',
      publishableKey: json['publishableKey']?.toString() ?? '',
    );
  }
}



